apps/examples/e2e/tests/test-kbds.spec.ts

```diff
@@ -0,0 +1,520 @@
+import test, { expect } from '@playwright/test'
+import { App } from '@tldraw/tldraw'
+import { setup, setupWithShapes } from '../shared-e2e'
+
+export function sleep(ms: number) {
+	return new Promise((resolve) => setTimeout(resolve, ms))
+}
+
+declare const app: App
+declare const __tldraw_event: { name: string }
+
+test.describe('keyboard shortcuts', () => {
+	test.beforeEach(setup)
+
+	test('tools', async ({ page }) => {
+		const geoToolKds = [
+			['r', 'rectangle'],
+			['o', 'ellipse'],
+		]
+
+		for (const [key, geo] of geoToolKds) {
+			await page.keyboard.press('v') // set back to select
+			await page.keyboard.press(key)
+			expect(await page.evaluate(() => app.currentToolId), `${key} -> ${geo}`).toBe('geo')
```

Comment:
Similar to a previous comment, would it make sense for us to check whether the correct tool is selected in the UI instead of checking the app's state?